mutters-core/src/main/java/com/rabidgremlin/mutters/core/IntentMatch.java

```diff
@@ -16,24 +17,32 @@
   private Intent intent;
 
   /** Map of slots that were matched. */
-  private HashMap<Slot, SlotMatch> slotMatches;
+  private Map<Slot, SlotMatch> slotMatches;
 
   /** The utterance that was matched against. */
   private String utterance;
 
   /**
-   * Constructor.
+   * The matching scores from the matcher. Should be set even if no match was
+   * found.
+   */
+  private MatcherScores matcherScores;
+
+  /**
+   * Constructor for the intent match. Use for a successful match.
    * 
-   * @param intent The intent that was matched.
-   * @param slotMatches The slots that were matched.
-   * @param utterance The utterance that was matched against.
+   * @param intent        The intent that was matched.
+   * @param slotMatches   The slots that were matched.
+   * @param utterance     The utterance that was matched against.
+   * @param matcherScores The scores generated by the matcher.
    */
-  public IntentMatch(Intent intent, HashMap<Slot, SlotMatch> slotMatches, String utterance)
+  public IntentMatch(Intent intent, Map<Slot, SlotMatch> slotMatches, String utterance, MatcherScores matcherScores)
```

Comment:
so if this holds `Intent` then it won't need `slotMatches` anymore?

Reply:
I wonder why they were always separate. Will investigate.

Reply:
AH, found it. The Intent holds the potential slots not the matched ones (which also have teh info on what was matched).